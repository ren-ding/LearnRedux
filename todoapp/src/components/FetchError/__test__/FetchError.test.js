import React from 'react';
import renderer from 'react-test-renderer';
import FetchError from '../FetchError';
import {BrowserRouter} from 'react-router-dom';

describe('FetchError',()=>{ 
    let props;

    beforeEach(()=>{
        props = {
            message: '',
            onRetry: ()=>{}
        }
    });

    describe('render',()=>{
        it('should render match with snapshot',()=> {
            const tree = renderer.create(
                <BrowserRouter>
                <FetchError {...props}/>
                </BrowserRouter>
            ).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});