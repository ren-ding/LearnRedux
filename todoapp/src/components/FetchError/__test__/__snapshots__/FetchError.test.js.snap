// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchError render should render match with snapshot 1`] = `
<div>
  <p>
    Could not fetch todos. 
    
  </p>
  <button
    onClick={[Function]}
  >
    Retry
  </button>
</div>
`;
